controllers/storagecluster/storagecluster_controller.go

```diff
@@ -211,6 +211,31 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
+	crdPredicate := predicate.Funcs{
+		CreateFunc: func(e event.TypedCreateEvent[client.Object]) bool {
+			crdAvailable, keyExist := r.AvailableCrds[e.Object.GetName()]
+			if keyExist && !crdAvailable {
```

Comment:
This does not make sense. This is a create event so it will be invoked in 1 of 2 cases:
1. The operator just started - which mean it is ok to queue an event
2. A CRD was created where it was not there before - which mean it is ok to queue an event

Reply:
We still need to check if the keyExist but we can remote the crdAvailable condition